app/questionnaire/questionnaire_manager.py

```diff
@@ -60,11 +60,11 @@ def update_questionnaire_store(self, location):
         # Store answers in QuestionnaireStore
         questionnaire_store = get_questionnaire_store(current_user.user_id, current_user.user_ik)
 
-        for answer in self.get_state_answers(location['block_id']):
+        for answer in self.get_state_answers(location.block_id):
             questionnaire_store.answer_store.add_or_update(answer.flatten())
 
-        if location not in questionnaire_store.completed_blocks:
-            questionnaire_store.completed_blocks.append(location)
+        if location.__dict__ not in questionnaire_store.completed_blocks:
```

Comment:
Maybe we should think about completed blocks being a list of Location objects, and when we serialise/deserialise in QuestionnaireStore we convert it to/from a dict.